### ISSUE TITLE
ITI: Allow `amp-list` to "shrink" when `src` is changed via `amp-bind`

### ISSUE THREAD

**Issue by aghassemi** [ampproject/amphtml#19939]
When `src` of `amp-list` is changed via `amp-bind`, the list will resize, however currently the list is only allowed to **expand** and it never **shrinks**.

See https://plain-punch.glitch.me/ to see this in action.

We intent to fix this behavior and allow `amp-list` to expand and shrink after its source is changed via `amp-bind`.

This change will initially be under an **experiment** flag that will be enabled in Canary and slowly will get pushed to production in January by ramping up the experiment from `1%` to `10%` to `50%` to `100%`. 

/cc @choumx @cathyxz @leonalicious

**Comment by aghassemi** [comment]
Update:

Unfortunately this happens to be a breaking change for certain usages of `amp-list` such as https://m.aliexpress.com/category/200001918/accessories.html?categoryName=Accessories&spm=a2g0n.category-amp.leafcat.Accessories where CSS does not expect an explicit `height` value to be added to the element. 

Moving this for V2 of `amp-list` where we can make breaking changes.

**Comment by aghassemi** [comment]
Meanwhile, the `changeToLayoutContainer()` action and its equivalent `[is-layout-container]` bindable attribute as implemented in https://github.com/ampproject/amphtml/issues/10230 can be used as an alternative in cases where list needs to shrink after a user-action.

**Comment by ampprojectbot** [comment]
This issue doesn't have a category which makes it harder for us to keep track of it. @aghassemi Please add an appropriate category.

**Comment by caroqliu** [comment]
Closing as this is resolved in v1 and planned for v2.
